### ISSUE TITLE
CPU training stalls beyond 60 s compiling native artifact for quantized attention and xgbst predictor kernels

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#496]
Public `recipe.train().epochs(1)` on `backend:cpu` remained active after 60 seconds. The earliest failure is `recipe.train().run()` stalling at `compile_model` -> `NativeModelIr::from_graph` -> `emit` -> `compile_native_artifact` on the CPU backend for the `.attn(1)` block and `.xgbst()` predictor. The `data/temporal/chronological_splits` loader feeds a small temporal row and feature count. The model operations, including 1 attention head and default XGBoost candidates (8 boosting iterations, depth 5), produce minimal arithmetic work and memory traffic that theoretically justify a sub-second CPU runtime. The measured 60 s is materially slower than hardware justifies; the dominant gap is the unreported CPU native LLVM compilation stall for Attention and XGBoost.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `97fb5b14c48e67ed`
- Classifier provider: `agy`
- Classifier model: `gpt-oss-120b-medium`
- Classifier effort: `medium`

### Classifier decision

```json
{"body":"Public `recipe.train().epochs(1)` on `backend:cpu` remained active after 60 seconds. The earliest failure is `recipe.train().run()` stalling at `compile_model` -> `NativeModelIr::from_graph` -> `emit` -> `compile_native_artifact` on the CPU backend for the `.attn(1)` block and `.xgbst()` predictor. The `data/temporal/chronological_splits` loader feeds a small temporal row and feature count. The model operations, including 1 attention head and default XGBoost candidates (8 boosting iterations, depth 5), produce minimal arithmetic work and memory traffic that theoretically justify a sub-second CPU runtime. The measured 60 s is materially slower than hardware justifies; the dominant gap is the unreported CPU native LLVM compilation stall for Attention and XGBoost.","issue":0,"rationale":"Searched open performance issues for CPU compilation stalls. While #279 covers Attention/Pool, #400 covers predictor/scan, and #403 covers Attention/Conv, there is no open issue for Attention paired with XGBoost estimators on CPU (AMD is covered in #171). The implementation path (Attention + XGBoost lowering to a CPU native artifact) and the compilation stall represent a new cause not covered by existing issues.","title":"CPU training stalls beyond 60 s compiling native artifact for quantized attention and xgbst predictor kernels","verdict":"new"}
```

## Deterministic failure packet

kind=performance
id=97fb5b14c48e67ed
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:14717 next:14718 composition:44896676233063
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:cpu
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-114-14717/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/chronological_splits")
		.target("target")
		.norm(z_score);
	let model = recipe.model()
		.attn(1).cos().iq(1).m
		.xgbst().elu().norm(batch).qi(4).k.m
		.loss(huber);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(44896676233063)
		.epochs(1)
		.log(all)
		.stop(0.0)
		.int(1)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=cpu

**Comment by nm-z** [comment]
Closing as a duplicate of #255. Both issues report CPU native-artifact compilation stalling when an XGBoost predictor is present. PR #550 verified this packet only on unmerged PR #385, so this closure does not claim that current `minimal` contains PR #385.
